src/objects/zcl_abapgit_object_bdef.clas.abap

```diff
@@ -485,4 +538,114 @@ CLASS zcl_abapgit_object_bdef IMPLEMENTATION.
 
   ENDMETHOD.
 
+
+  METHOD get_object_data.
+
+    DATA:
+      lo_object_data TYPE REF TO object,
+      lr_metadata    TYPE REF TO data,
+      lr_data        TYPE REF TO data.
+
+    FIELD-SYMBOLS:
+      <lv_metadata_node> TYPE any,
+      <ls_metadata>      TYPE any,
+      <lv_source>        TYPE any,
+      <lg_data>          TYPE any.
+
+    CREATE DATA lr_data TYPE ('CL_BLUE_SOURCE_OBJECT_DATA=>TY_OBJECT_DATA').
+    ASSIGN lr_data->* TO <lg_data>.
+    ASSERT sy-subrc = 0.
+
+    ASSIGN COMPONENT 'METADATA' OF STRUCTURE <lg_data> TO <lv_metadata_node>.
+    ASSERT sy-subrc = 0.
+
+    CREATE DATA lr_metadata  TYPE ('IF_ADT_TOOLS_CORE_SOURCE_TYPES=>TY_ABAP_SOURCE_MAIN_OBJECT').
+    ASSIGN lr_metadata->* TO <ls_metadata>.
+    ASSERT sy-subrc = 0.
+
+    io_xml->read(
+      EXPORTING
+        iv_name = 'BDEF'
+      CHANGING
+        cg_data = <ls_metadata> ).
+
+    <lv_metadata_node> = <ls_metadata>.
+
+    ASSIGN COMPONENT 'CONTENT-SOURCE' OF STRUCTURE <lg_data> TO <lv_source>.
+    ASSERT sy-subrc = 0.
+
+    <lv_source> = mo_files->read_string( 'asbdef' ).
+
+    CREATE OBJECT ro_object_data TYPE ('CL_BLUE_SOURCE_OBJECT_DATA').
+
+    CALL METHOD ro_object_data->('IF_WB_OBJECT_DATA_MODEL~SET_DATA')
+      EXPORTING
+        p_data = <lg_data>.
```

Comment:
@ThomasPloski 
still get this one
![image](https://user-images.githubusercontent.com/17437789/99276023-803bb200-282c-11eb-903c-78b43a4a6e93.png)